create a GitHub Action workflow to build and publish an EXE file

LovingThresh opened the pull request:
## What's Changed

Added a GitHub Actions workflow to automatically build and publish Windows executables:

- Runs daily at 00:00 UTC
- Creates date-based versioned releases (e.g., v2025.03.20)
- Works on Windows runners for the entire process
- Generates SHA256 checksums for verification

## Why

- Provides ready-to-use executables for users without building from source
- Eliminates manual release process
- Ensures daily availability of latest builds

## Implementation

- Single Windows workflow using PowerShell
- Properly handles file uploads and release management
- Automatically replaces previous day's release

GreatV reviewed (commented):
这个GitHub action更适用于 nightly build。建议做2个工作流， 将 nightly build 放在单独的工作流文件中，与正式发布区分开。正式发布只在每次打tag时运行。对于nightly build 建议 用单一固定的 "nightly" 标签每次构建会替换之前的 nightly 版本，通过 --prerelease 标记，明确标识为非正式版本。另外建议将中文注释改为英文注释。

LovingThresh commented:
## Improved GitHub Actions Workflow for Automated Builds
### What's Changed
Split the build process into two separate workflows for better organization and clarity:
#### 1. Nightly Build Workflow

Added a dedicated workflow for automatic nightly builds
Runs daily at 00:00 UTC and on pushes to main/master
Creates a permanent "nightly" release tag marked as pre-release
Updates the same nightly release with each new build
Includes date in filename (e.g., PPOCRLabel-nightly-20250320.exe)
Generates SHA256 checksums for verification

#### 2. Release Build Workflow

Added a separate workflow for official releases
Triggered only when version tags are pushed (e.g., v1.0.0)
Creates permanent version-based releases
Names executable files according to the version tag
Generates SHA256 checksums for verification

GreatV commented on .github/workflows/nightly-build.yml:
```diff
@@ -0,0 +1,98 @@
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          gh release upload nightly (Join-Path -Path "dist" -ChildPath "SHA256SUMS.txt") --repo ${{ github.repository }} --clobber
```
```suggestion
          gh release upload nightly (Join-Path -Path "dist" -ChildPath "SHA256SUMS.txt") --repo ${{ github.repository }} --clobber

```

GreatV reviewed (approved):
LGTM

GreatV commented:
你这是ai生成的吧，跑起来有问题。

LovingThresh commented:
Sorry, let me deal with it. What is the specific question?

LovingThresh commented:
I sincerely apologize. When pushing the code today at the office, I relied on my memory and inadvertently missed the **requirements.txt** file. I also later realized that the generated package size is **different from** the version I successfully built at home. But, most of the code itself is fine. I am still very sorry for this oversight.

GreatV commented:
Thanks for your contributions and please feel free to continue your contributions to the PPOCRLabel and the PaddleOCR community.

GreatV commented:
Daily builds may be too frequent, we need to delete this action and only keep the builds during releases.

GreatV commented:
Can you submit a PR to delete this?
